GetBrokerRedirectURI.ps1

```diff
@@ -0,0 +1,103 @@
+<#
+.SYNOPSIS
+Obtain the RedirectURI from an APK.  Requires JDK\bin\keytool and AndroidSDK\build-tools\###\aapt to be on the path.
+
+.EXAMPLE
+GetBrokerRedirectURI MyApp.apk
+#>
+Param(
+[Parameter(Mandatory=$true)] 
+[string]$apkFile
+)
+
+<#
+Extract all of the signature owner and SHA1 hex bytes from `keytool -printcert -jarfile`.
+#>
+function GetSHA1Hex {
+    Param(
+        [string]$apkFile
+        )
+    $output = keytool -printcert -jarfile $apkFile
+    $ownerPrefix = "Owner:"
+    $shaPrefix = "SHA1:"
+    $signatures = @{}
+    $owner = ""
+    ForEach ($line in $output) {
+        if ($line.contains($ownerPrefix)) {
+            $owner = $line.replace($ownerPrefix, "").trim()
+        }
+
+        if ($line.contains($shaPrefix)) {
+            $base64 = GetBase64FromSHA1Hex($line.replace($shaPrefix, "").trim())
+            $signatures.add($owner, $base64)
+        }
+    }
+    
+    if ($signatures.Keys.count -eq 0) {
+        write-error "No signatures found."
+        exit
+    }
+    
+    if ($signatures.Keys.count -gt 1) {
+        write-warning "Multiple signatures detected, please select the correct one."
+    }
+
+    return $signatures
+}
+
+<#
+Convert the string hex to bytes.
+#>
+function GetBase64FromSHA1Hex {
+    Param(
+        [string]$hexSig
+        )
+
+    $hex = $hexSig.split(":")
+    $bytes = @()
+    ForEach ($byte in $hex) {
+        $bytes += [convert]::ToByte($byte, 16)
+    }
+
+    [System.Convert]::ToBase64String($bytes)
+}
+
+<#
+Extract the APK package name from `aapt dump badging`.
+#>
+function GetPackageName {
+    Param(
+        [string]$apkFile
+    )
+
+    $output = aapt dump badging $apkFile
```

Comment:
What assumptions is the script making about the availability of tools to the shell?  You may want to add the following:
`get-command aapt
if(!$?){
 write-error "The appt command was not found.  Please add your android build tools to the path"
}
`
$? is an automatic variable which indicates whether the last command was successful or not.

Reply:
Adding `VerifyDependencies` function that checks if get-command is successful